app/utils/sparse-fieldsets.ts

```diff
@@ -0,0 +1,70 @@
+import { camelize, underscore } from '@ember/string';
+import ModelRegistry from 'ember-data/types/registries/model';
+import { pluralize, singularize } from 'ember-inflector';
+import { Links } from 'jsonapi-typescript';
+
+import { Resource } from 'osf-api';
+
+// Automatic type safety for attributes and relationships appears prohibitively expensive.
+// Instead, use `unknown` to force explicit types.
+export type SparseModel = {
+    id: string | number;
+    modelName: string;
+    links?: Links;
+} & Record<string, unknown>;
+
+export type SparseFieldset = {
+    [K in keyof ModelRegistry]?: string[]
+};
+
+function mapObject<T, R>(
+    obj: T,
+    mapKey: (k: string) => string,
+    mapValue: (v: T[keyof T]) => R,
+): Record<string, R> {
+    return Object.entries(obj).reduce(
+        (acc, [k, v]) => ({
+            ...acc,
+            [mapKey(k)]: mapValue(v),
+        }),
+        {} as Record<string, R>,
+    );
+}
+
+function extractEmbeds(resource: Resource): Record<string, SparseModel | SparseModel[]> {
+    return mapObject(
+        resource.embeds || {},
+        camelize,
```

Comment:
Why are you passing an import as a parameter to a function?

Reply:
I needed to do two things with `embeds` here:
- camelize the keys
- recursively parse the embedded resources

Would it be clearer to pass `key => camelize(key)`?

Reply:
yea actually that would be a lot clearer yea? Take a key, camelize that key is how that reads.